flang/test/Driver/ctofortran.f90

```diff
@@ -1,4 +1,5 @@
-! UNSUPPORTED: system-windows
+! MacOS needs -isysroot <osx_sysroot> with clang and flang to build binaries.
```

Comment:
Didn't you manage to work around this with changes in flang/test/lit.cfg.py?

Reply:
I managed to work around this when clang is used. The problem is that flang doesn't support the `-isysroot` flag.

If `-L<osx_sysroot>/usr/lib` is used with flang then it can output a linked binary, but I think it would be better to make flang also support `-isysroot`. I'll try to add support for it in a separate patch.

Reply:
So, IIUC, the changes in the cfg file are not used?

Reply:
They are not used in this test, but are used in `flang/test/Runtime/no-cpp-dep.c`. And now any flang test using `%cc` will already have the proper sysroot flag on MacOS.

After `-isysroot` support is added to flang, it will be possible to use the cfg changes here too.

Reply:
OK, makes sense.

Please note that right now this change is completely unrelated to what this patch is trying to achieve (from the summary):

> Skip unsupported flang tests on MacOS.

You can fix that by either: 
* updating the summary (to make it clear to our future selves that additional improvement is intended), or * by sending this change as a separate patch. 

Whatever works better for you :) I just want to make sure that there's a clear "why" behind every change in a commit summary (folks rely heavily on that kind of documentation).

Reply:
Sorry, I meant the change in flang/test/lit.cfg.py specifically. My bad for not being clearer!

Please go ahead and merge this. I am happy for you to leave the summary as is or update it before merging. No need for me to take another look.

I really appreciate you seeing this through 🙏🏻 !

Reply:
Thank you for reviewing it and for the helpful comments!